src/ide_rigid.cpp

```diff
@@ -89,6 +90,11 @@ void IDERigidDevice::post_image_setup()
         found_chs = true;
         method = "INI config";
     }
+    else if (((IDEImageFile*)(m_image))->direct_file()->setCHS(m_devinfo.cylinders, m_devinfo.heads, m_devinfo.sectors_per_track))
```

Comment:
Instead of the cast that may crash in weird ways, I think it would be better to add `direct_file()` to `IDEImage` interface. It could then return `null` if we ever have non-file images. This code should then be `else if (m_image->direct_file() && m_image->direct_file()->setCHS(...)`.